src/syft/core/common/module.py

```diff
@@ -77,3 +83,81 @@ def cpu(self) -> "Module":
         for _, module in self.modules.items():
             module.cpu()
         return self
+
+    # zero them so we know they are copied
+    def zero_layers(self) -> None:
+        for _, m in self.modules.items():
+            if hasattr(m, "weight"):
+                m.weight.requires_grad_(False).zero_()
+            if hasattr(m, "bias"):
+                m.bias.requires_grad_(False).zero_()
+
+    # easy way to check the weights have changed
+    def sum_layers(self) -> None:
+        for n, m in self.modules.items():
+            if hasattr(m, "state_dict"):
+                for k, v in m.state_dict().items():
+                    if hasattr(v, "sum"):
+                        s = v.sum().item()
+                        log = f"> Layer {n} sum({k}): {s}"
+                        print(log)
+                        logger.debug(log)
+
+    # copy the weights, there are blocking requests so make sure to respond
+    def copy_remote_state(
+        self,
+        remote_model: Any,
+        request_name: str,
+        reason: str,
+        timeout_secs: int,
+        delete_obj: bool = False,
+        skip_layers: List[str] = [],
+    ) -> None:
+        # loop over models module pointers
+        for n, m in remote_model.modules.items():
+            try:
+                if n in skip_layers:
+                    log = f"> Skipping: {n}"
+                    print(log)
+                    logger.debug(log)
+                    continue
+                if hasattr(self, n):
+                    # get the equivalent layer for the local model
+                    local_m = getattr(self, n)
+
+                    # if the remote layer has a state_dict
+                    if hasattr(m, "state_dict"):
+                        sd_ptr = m.state_dict()
+                        # get a blocking copy of the state_dict
+                        log = f"> Downloading remote: {n}"
+                        print(log)
+                        logger.debug(log)
+                        state_dict = sd_ptr.get(
+                            request_block=True,
+                            request_name=request_name,
+                            reason=reason,
+                            timeout_secs=timeout_secs,
+                            delete_obj=delete_obj,
+                        )
+                        # iterate through the key, values
+                        # weights and biases should be in there
+                        for key, value in state_dict.items():
+                            key_str = key.upcast()
+                            # if the local models module has the same property
+                            # e.g. .weight or .bias
+                            if hasattr(local_m, key_str):
+                                # if the downloaded value is not a Parameter
+                                # its a tensor so we need to convert it
+                                if not issubclass(type(value), torch.nn.Parameter):
```

Comment:
Q: Can't we use
```if not isinstance(valu,e torch.nn.Parameter)```?

Reply:
I wasn't sure if there were cases where nn.Parameter might get subclassed. Maybe we should be changing the part which gives us Tensors instead of Parameters, but i think thats blocked behind the availability of methods and serialization for all the module allowlist methods. For now, I have updated this to `isinstance` as per your suggestion.